### ISSUE TITLE
LLVM ERROR: Global variable 'security_txt' has an invalid section specifier '.security.txt'

### ISSUE THREAD

**Issue by jonorally** [neodyme-labs/solana-security-txt#3]
After adding a `security_txt!` in my src/lib.rs, `cargo build` and `cargo test` begin to fail with this error:

>`LLVM ERROR: Global variable 'security_txt' has an invalid section specifier '.security.txt': mach-o section specifier requires a segment and section separated by a comma.`

`cargo +bpf build --target bpfel-unknown-unknown --release` (which is being called by `anchor build`) does seem to work without issues even with the macro left in though

Copying the macro directly into the code and removing the `#[link_section]` line also works fine

```rust
#[macro_export]
macro_rules! security_txt {
    ($($name:ident: $value:expr),*) => {
        #[allow(dead_code)]
        #[no_mangle]
        // #[link_section = ".security.txt"] // removing this works
        pub static security_txt: &str = concat! {
            "=======BEGIN SECURITY.TXT V1=======\0",
             $(stringify!($name), "\0", $value, "\0",)*
            "=======END SECURITY.TXT V1=======\0"
        };
    };
}
```

**Comment by tlambertz** [comment]
What OS are you using? From `mach-o` in the error I am inferring macos?

**Comment by jonorally** [comment]
Yup macos, here's this if it helps

```
$ sw_vers
ProductName:	macOS
ProductVersion:	12.3.1
BuildVersion:	21E258
$ uname -mrs
Darwin 21.4.0 x86_64
```

**Comment by tlambertz** [comment]
Yeah there seems to be an issue with the section name if building natively for mac, as you would for tests.

We didn't test with mac's beforehand. I'll think about how to best fix it. Might just drop the section thing all-together. It's kinda nice to have, but not when it breaks things. Or we could only include it when the target is eBPF.

Thanks for the report!

**Comment by jonorally** [comment]
No problem, thanks for the quick response! 

If I'm reading the code comments correctly, it's fine to just comment out the `#[link_section]` since it's optional in the standard anyway? Would prefer doing that rather than manually removing/adding the whole security_txt back in when I build natively vs for release

**Comment by tlambertz** [comment]
Yes, feel free to remove it.

**Comment by CherryWorm** [comment]
Fixed in b08a4316cca2f86f977cb0b13fc57bdc93406227
